core-overlay.html

```diff
@@ -111,666 +124,373 @@
-          old.style.position = '';
-          old.style.outline = '';
+        this.__overlaySetup = true;
+        var style = this.style;
+        style.outline = 'none';
+        style.display = 'none';
+      },
+
+      openedChanged: function() {
+        if (this._openChangedAsync) {
+          this.cancelAsync(this._openChangedAsync);
         }
-        old.style.display = '';
-      }
-    },
 
-    transitionChanged: function(old) {
-      if (!this.target) {
-        return;
-      }
-      if (old) {
-        this.getTransition(old).teardown(this.target);
-      }
-      this.target.__overlaySetup = false;
-    },
-
-    // NOTE: wait to call this until we're as sure as possible that target
-    // is styled.
-    ensureTargetSetup: function() {
-      if (!this.target || this.target.__overlaySetup) {
-        return;
-      }
-      if (!this.sizingTarget) {
-        this.sizingTarget = this.target;
-      }
-      this.target.__overlaySetup = true;
-      this.target.style.display = '';
-      var transition = this.getTransition();
-      if (transition) {
-        transition.setup(this.target);
-      }
-      var style = this.target.style;
-      var computed = getComputedStyle(this.target);
-      if (computed.position === 'static') {
-        style.position = 'fixed';
-      }
-      style.outline = 'none';
-      style.display = 'none';
-    },
-
-    openedChanged: function() {
-      this.transitioning = true;
-      this.ensureTargetSetup();
-      this.prepareRenderOpened();
-      // async here to allow overlay layer to become visible.
-      this.async(function() {
-        this.target.style.display = '';
-        // force layout to ensure transitions will go
-        this.target.offsetWidth;
-        this.renderOpened();
-      });
-      this.fire('core-overlay-open', this.opened);
-    },
-
-    // tasks which must occur before opening; e.g. making the element visible
-    prepareRenderOpened: function() {
-      if (this.opened) {
-        addOverlay(this);
-      }
-      this.prepareBackdrop();
-      // async so we don't auto-close immediately via a click.
-      this.async(function() {
-        if (!this.autoCloseDisabled) {
-          this.enableElementListener(this.opened, document,
-              this.captureEventName, 'captureHandler', true);
+        this.transitioning = true;
+
+        this.ensureTargetSetup();
+        this.prepareRenderOpened();
+
+        // async here to allow overlay layer to become visible.
+        this._openChangedAsync = this.async(function() {
+          this.style.display = '';
+          // force layout to ensure transitions will go
+          this.offsetWidth;
+          this.renderOpened();
+          this._openChangedAsync = null;
+        });
+
+        this.fire('core-overlay-open', {
+          opened: this.opened
+        });
+      },
+
+      // tasks which must occur before opening; e.g. making the element visible
+      prepareRenderOpened: function() {
```

Comment:
This method has some fairly complex conditional behavior. Is it reasonable to refactor this so that the expected outcome for cases when `opened` is `true` or `false` is more clear?